File: 프로그래머스/2/42888. 오픈채팅방/오픈채팅방.py
def solution(record):
    answer = []
    d = {}
    for r in record:
        rs = r.split()
        if rs[0] in ["Enter", "Change"]:
            d[rs[1]] = rs[2]
    for r in record:
        rs = r.split()
        if rs[0] == "Enter":
            answer.append(d[rs[1]]+"님이 들어왔습니다.")
        elif rs[0] == "Leave":
            answer.append(d[rs[1]]+"님이 나갔습니다.")
    return answer

File: 프로그래머스/2/42888. 오픈채팅방/오픈채팅방.js
function solution(record) {
    let answer = [];
    let d = {};
    for (let r of record) {
        let rs = r.split(" ");
        if (rs[0] === "Enter" || rs[0] === "Change") {
            d[rs[1]] = rs[2];  // 사용자 ID에 해당하는 닉네임 저장
        }
    }
    for (let r of record) {
        let rs = r.split(" ");
        if (rs[0] === "Enter") {
            answer.push(d[rs[1]] + "님이 들어왔습니다.");
        } else if (rs[0] === "Leave") {
            answer.push(d[rs[1]] + "님이 나갔습니다.");
        }
    }

    return answer;
}
